aidial_assistant/application/assistant_application.py

```diff
@@ -103,17 +115,14 @@ async def chat_completion(
                 or info.ai_plugin.description_for_human
             )
 
-        usage_publisher = UsagePublisher()
         command_dict: CommandDict = {
-            RunPlugin.token(): lambda: RunPlugin(model, tools, usage_publisher),
+            RunPlugin.token(): lambda: RunPlugin(model, tools),
             Reply.token(): Reply,
         }
         chain = CommandChain(
-            model_client=model,
-            name="ASSISTANT",
-            command_dict=command_dict,
-            usage_publisher=usage_publisher,
+            model_client=model, name="ASSISTANT", command_dict=command_dict
         )
+        _validate_messages(request.messages)
```

Comment:
Let's move all kinds of validation checks (`_validate_messages`, addons check) as close to the begging of the method as possible; to make them first to trigger, before any other kinds of exceptions.